// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: module/data_obj.proto

package module

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on DataObj with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *DataObj) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on DataObj with the rules defined in the
// proto definition for this message. If any rules are violated, the result is
// a list of violation errors wrapped in DataObjMultiError, or nil if none found.
func (m *DataObj) ValidateAll() error {
	return m.validate(true)
}

func (m *DataObj) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Id

	// no validation rules for Uuid

	// no validation rules for Name

	// no validation rules for Type

	// no validation rules for Desc

	// no validation rules for Size

	// no validation rules for Status

	// no validation rules for UserId

	// no validation rules for Version

	// no validation rules for CreateTime

	// no validation rules for UpdateTime

	// no validation rules for DeleteTime

	if len(errors) > 0 {
		return DataObjMultiError(errors)
	}

	return nil
}

// DataObjMultiError is an error wrapping multiple validation errors returned
// by DataObj.ValidateAll() if the designated constraints aren't met.
type DataObjMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m DataObjMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m DataObjMultiError) AllErrors() []error { return m }

// DataObjValidationError is the validation error returned by DataObj.Validate
// if the designated constraints aren't met.
type DataObjValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e DataObjValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e DataObjValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e DataObjValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e DataObjValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e DataObjValidationError) ErrorName() string { return "DataObjValidationError" }

// Error satisfies the builtin error interface
func (e DataObjValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sDataObj.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = DataObjValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = DataObjValidationError{}
